In [92]:
# Exploratory Data Analysis (EDA)

import pandas as pd
from pathlib import Path
import plotly.express as px
import plotly.graph_objects as go
import numpy as np


root = Path.cwd().parent
print(root)

c:\Users\sebas\PycharmProjects\Git\Seb_branch\institutional-roi-analysis


In [93]:
df = pd.read_csv(root/"data"/"raw"/"scorecard"/"scorecard_FL_programs.csv")
df.head()

code                                              title  unit_id  distance  \
0  3906               Theological and Ministerial Studies.   132408         3   
1  1101        Computer and Information Sciences, General.   132471         2   
2  1304        Educational Administration and Supervision.   132471         2   
3  2201                                               Law.   132471         1   
4  2401  Liberal Arts and Sciences, General Studies and...   132471         1   

          school.type  school.main_campus  credential.level  \
0  Private, nonprofit                   1                 3   
1  Private, nonprofit                   1                 3   
2  Private, nonprofit                   1                 5   
3  Private, nonprofit                   1                 7   
4  Private, nonprofit                   1                 3   

   earnings.4_yr.overall_median_earnings  \
0                                31484.0   
1                                52714.0   
2                                69477.0   
3                                62176.0   
4                                40411.0   

   earnings.4_yr.working_not_enrolled.overall_count  \
0                                              36.0   
1                                              66.0   
2                                              87.0   
3                                             197.0   
4                                              36.0   

   debt.staff_grad_plus.all.eval_inst.median  \
0                                    23946.0   
1                                    27750.0   
2                                    51250.0   
3                                   154477.0   
4                                    38125.0   

   debt.staff_grad_plus.all.eval_inst.median_payment  \
0                                              254.0   
1                                              294.0   
2                                              583.0   
3                                             1757.0   
4                                              404.0   

                     school.name  
0  Baptist University of Florida  
1               Barry University  
2               Barry University  
3               Barry University  
4               Barry University

In [94]:
df.isna().sum()

code                                                 0
title                                                0
unit_id                                              0
distance                                             0
school.type                                          0
school.main_campus                                   0
credential.level                                     0
earnings.4_yr.overall_median_earnings                0
earnings.4_yr.working_not_enrolled.overall_count     0
debt.staff_grad_plus.all.eval_inst.median            0
debt.staff_grad_plus.all.eval_inst.median_payment    0
school.name                                          0
dtype: int64

In [95]:
df.duplicated().sum()

np.int64(0)

In [96]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1538 entries, 0 to 1537
Data columns (total 12 columns):
 #   Column                                             Non-Null Count  Dtype  
---  ------                                             --------------  -----  
 0   code                                               1538 non-null   int64  
 1   title                                              1538 non-null   str    
 2   unit_id                                            1538 non-null   int64  
 3   distance                                           1538 non-null   int64  
 4   school.type                                        1538 non-null   str    
 5   school.main_campus                                 1538 non-null   int64  
 6   credential.level                                   1538 non-null   int64  
 7   earnings.4_yr.overall_median_earnings              1538 non-null   float64
 8   earnings.4_yr.working_not_enrolled.overall_count   1538 non-null   float64
 9   debt.staff_grad_plu

In [97]:
df.describe()

code        unit_id     distance  school.main_campus  \
count  1538.000000    1538.000000  1538.000000         1538.000000   
mean   3732.440182  219304.497399     1.475293            0.809493   
std    1693.623708  140554.436996     0.812861            0.392829   
min     100.000000  132408.000000     0.000000            0.000000   
25%    1609.000000  133997.000000     1.000000            1.000000   
50%    4653.000000  136215.000000     1.000000            1.000000   
75%    5135.000000  385619.000000     2.000000            1.000000   
max    5401.000000  497222.000000     3.000000            1.000000   

       credential.level  earnings.4_yr.overall_median_earnings  \
count       1538.000000                            1538.000000   
mean           3.017555                           51970.442133   
std            1.387866                           22434.632713   
min            1.000000                           12467.000000   
25%            2.000000                           36910.000000   
50%            3.000000                           48033.500000   
75%            3.000000                           64225.250000   
max            8.000000                          192498.000000   

       earnings.4_yr.working_not_enrolled.overall_count  \
count                                       1538.000000   
mean                                         159.837451   
std                                          368.124786   
min                                           16.000000   
25%                                           36.000000   
50%                                           68.000000   
75%                                          155.000000   
max                                         6804.000000   

       debt.staff_grad_plus.all.eval_inst.median  \
count                                1538.000000   
mean                                27568.488947   
std                                 29789.610146   
min                                  3167.000000   
25%                                 15000.000000   
50%                                 20000.000000   
75%                                 29247.250000   
max                                406377.000000   

       debt.staff_grad_plus.all.eval_inst.median_payment  
count                                        1538.000000  
mean                                          302.143043  
std                                           340.883162  
min                                            34.000000  
25%                                           159.000000  
50%                                           212.000000  
75%                                           317.500000  
max                                          4623.000000

Initial inspection confirms no missing values or duplicates. The dataset contains 1,538 program-level observations across Florida institutions.

How concentrated are earnings and debt outcomes across programs?

In [98]:
px.histogram(df, x="earnings.4_yr.overall_median_earnings")

Median earnings appear right-skewed, suggesting the presence of high-earning outliers. Log transformation is applied to stabilize variance.

Does log transformation improve comparability across programs?

In [99]:
px.histogram(
    df,
    x="debt.staff_grad_plus.all.eval_inst.median_payment"
)

In [100]:
px.histogram(df, x=np.log1p(df["earnings.4_yr.overall_median_earnings"]))

In [101]:
px.histogram(df, x=np.log1p(df["debt.staff_grad_plus.all.eval_inst.median_payment"]))

Debt and debt payment distributions exhibit similar skewness, reinforcing the need for transformation prior to modeling.

In [102]:
dfx = df
dfx["log_median_earnings"]=np.log1p(dfx["earnings.4_yr.overall_median_earnings"])
dfx["log_median_debt_payment"]=np.log1p(dfx["debt.staff_grad_plus.all.eval_inst.median_payment"]*12)
dfx["log_median_debt"]=np.log1p(dfx["debt.staff_grad_plus.all.eval_inst.median"])

In [103]:
earning_count = go.Figure(
    go.Histogram(
        x=df["earnings.4_yr.working_not_enrolled.overall_count"],
        xbins=dict(start=0, end=6804, size=30)
    )
)

earning_count.show()


What is an appropriate threshold to remove low graduate programs?

In [104]:
dfx = dfx.loc[df["earnings.4_yr.working_not_enrolled.overall_count"] >= 25].reset_index()
lt50=len(df) - len(dfx)
lt50

190

Programs with fewer than 25 observed earners are excluded to reduce instability in earnings estimates.

In [105]:
summary_overview = pd.DataFrame({
    "Total Programs": [len(df)],
    "After ≥25 student Threshold": [len(dfx)],
    "Mean Earnings": [dfx["earnings.4_yr.overall_median_earnings"].mean()],
    "Median Earnings": [dfx["earnings.4_yr.overall_median_earnings"].median()],
    "Median Debt": [dfx["debt.staff_grad_plus.all.eval_inst.median"].median()],
})
summary_overview


Total Programs  After ≥25 student Threshold  Mean Earnings  \
0            1538                         1348   51775.038576   

   Median Earnings  Median Debt  
0          48033.5      19783.5

Filtering reduces the sample from 1,538 to 1,348 programs (about 12% removed).

Debt burden approximates the share of annual earnings required to service debt obligations.

How many programs exhibit unusually high financial stress relative to earnings?

In [106]:
dfx["debt_burden"] = (dfx["debt.staff_grad_plus.all.eval_inst.median_payment"]*12)/dfx["earnings.4_yr.overall_median_earnings"]
(dfx["debt_burden"]>=0.25).sum()

np.int64(21)

21 programs require at least 25% of annual earnings to service debt, indicating extreme financial stress.

In [107]:
(dfx["debt_burden"]<=0.02).sum()

np.int64(32)

32 programs require 2% or less of earnings, representing unusually low financial burden.

In [108]:
px.histogram(
    dfx,
    x="debt_burden"
)

In [109]:
dfx["debt_burden"].describe(percentiles=[.1,.25,.5,.75,.9])

count    1348.000000
mean        0.067795
std         0.048653
min         0.009007
10%         0.029909
25%         0.039711
50%         0.054346
75%         0.079315
90%         0.114919
max         0.483266
Name: debt_burden, dtype: float64

In [110]:
dfx["log_debt_burden"] = np.log1p(dfx["debt_burden"])

The median program requires about 5.4% of earnings to service debt. However, the upper tail extends sharply, with the 90th percentile approaching 11.5% and extreme cases near 48%.

Is financial stress concentrated in specific credential levels or institutional types?

In [111]:
dfx["earnings_to_debt_ratio"] = (
    dfx["earnings.4_yr.overall_median_earnings"] /
    dfx["debt.staff_grad_plus.all.eval_inst.median"]
)

fltop10 = dfx.sort_values("earnings_to_debt_ratio", ascending=False)["title"].head(10)
dfx.sort_values("earnings_to_debt_ratio", ascending=False).head(10)


index  code                                              title  unit_id  \
706    815  4301                  Criminal Justice and Corrections.   136358   
707    816  5109  Allied Health Diagnostic, Intervention, and Tr...   136358   
88     114  4301                  Criminal Justice and Corrections.   133386   
536    615  5138  Registered Nursing, Nursing Administration, Nu...   134608   
91     117  5109  Allied Health Diagnostic, Intervention, and Tr...   133386   
525    604  5109  Allied Health Diagnostic, Intervention, and Tr...   134495   
591    679  5138  Registered Nursing, Nursing Administration, Nu...   135717   
32      39  5138  Registered Nursing, Nursing Administration, Nu...   132709   
579    666  5138  Registered Nursing, Nursing Administration, Nu...   135160   
521    599  5138  Registered Nursing, Nursing Administration, Nu...   134343   

     distance school.type  school.main_campus  credential.level  \
706         1      Public                   1                 1   
707         1      Public                   1                 1   
88          1      Public                   1                 1   
536         3      Public                   1                 3   
91          1      Public                   1                 1   
525         1      Public                   1                 1   
591         1      Public                   1                 2   
32          1      Public                   1                 2   
579         1      Public                   1                 2   
521         1      Public                   1                 2   

     earnings.4_yr.overall_median_earnings  \
706                                69281.0   
707                                50192.0   
88                                 49543.0   
536                                78190.0   
91                                 51952.0   
525                                49488.0   
591                                73314.0   
32                                 68951.0   
579                                63996.0   
521                                62103.0   

     earnings.4_yr.working_not_enrolled.overall_count  \
706                                             139.0   
707                                              75.0   
88                                              144.0   
536                                              52.0   
91                                               52.0   
525                                              63.0   
591                                             140.0   
32                                               95.0   
579                                              89.0   
521                                             110.0   

     debt.staff_grad_plus.all.eval_inst.median  \
706                                     4892.0   
707                                     4025.0   
88                                      4446.0   
536                                     7250.0   
91                                      5500.0   
525                                     5500.0   
591                                     9500.0   
32                                      9000.0   
579                                     8440.0   
521                                     8239.0   

     debt.staff_grad_plus.all.eval_inst.median_payment  \
706                                               52.0   
707                                               43.0   
88                                                47.0   
536                                               77.0   
91                                                58.0   
525                                               58.0   
591                                              101.0   
32                                                95.0   
579                                               89.0   
521                                               87.0   

                        school.name  log_median_earnings  \
7

High earnings-to-debt ratios are concentrated in applied public programs such as Nursing and Criminal Justice.

In [112]:
flbott_10 = dfx.sort_values("earnings_to_debt_ratio", ascending=True)["title"].head(10)
dfx.sort_values("earnings_to_debt_ratio", ascending=True).head(10)

index  code                                              title  unit_id  \
673     776  5104                                         Dentistry.   136215   
13       17  5112                                          Medicine.   132471   
654     756  1310                    Special Education and Teaching.   136215   
297     353  5112                                          Medicine.   133951   
3         3  2201                                               Law.   132471   
608     704  5112                                          Medicine.   135726   
296     352  5109  Allied Health Diagnostic, Intervention, and Tr...   133951   
677     780  5112                                          Medicine.   136215   
858     990  2201                                               Law.   137476   
1086   1266  2201                                               Law.   442295   

      distance         school.type  school.main_campus  credential.level  \
673          0  Private, nonprofit                   1                 6   
13           1  Private, nonprofit                   1                 7   
654          3  Private, nonprofit                   1                 5   
297          1              Public                   1                 7   
3            1  Private, nonprofit                   1                 7   
608          1  Private, nonprofit                   1                 7   
296          1              Public                   1                 5   
677          0  Private, nonprofit                   1                 6   
858          0  Private, nonprofit                   1                 6   
1086         1  Private, nonprofit                   1                 7   

      earnings.4_yr.overall_median_earnings  \
673                                114794.0   
13                                 106396.0   
654                                 48527.0   
297                                 82512.0   
3                                   62176.0   
608                                 83257.0   
296                                 49226.0   
677                                143354.0   
858                                 71477.0   
1086                                69213.0   

      earnings.4_yr.working_not_enrolled.overall_count  \
673                                              175.0   
13                                                88.0   
654                                               62.0   
297                                               67.0   
3                                                197.0   
608                                              271.0   
296                                               39.0   
677                                              377.0   
858                                              296.0   
1086                                             116.0   

      debt.staff_grad_plus.all.eval_inst.median  \
673                                    406377.0   
13                                     278906.0   
654                                    125828.0   
297                                    207971.0   
3                                      154477.0   
608                                    202806.0   
296                                    117918.0   
677                                    336670.0   
858                                    166022.0   
1086                                   144259.0   

      debt.staff_grad_plus.all.eval_inst.median_payment  \
673                                              4623.0   
13                                               3173.0   
654                                              1431.0   
297                                              2366.0   
3                                                1757.0   
608                                              2307.0   
296                                              1341.0   
677                                              3830.0   
858                        

Low earnings-to-debt ratios are concentrated in professional programs (Law, Medicine, Dentistry), reflecting high borrowing levels relative to early-career earnings.

In [113]:
dfx.sort_values("earnings_to_debt_ratio", ascending=True)[df["credential.level"]<6].head(10)

C:\Users\sebas\AppData\Local\Temp\ipykernel_9924\1564741994.py:1: UserWarning:

Boolean Series key will be reindexed to match DataFrame index.



index  code                                              title  unit_id  \
673     776  5104                                         Dentistry.   136215   
13       17  5112                                          Medicine.   132471   
654     756  1310                    Special Education and Teaching.   136215   
297     353  5112                                          Medicine.   133951   
608     704  5112                                          Medicine.   135726   
296     352  5109  Allied Health Diagnostic, Intervention, and Tr...   133951   
677     780  5112                                          Medicine.   136215   
858     990  2201                                               Law.   137476   
1086   1266  2201                                               Law.   442295   
1070   1249  5133  Alternative and Complementary Medicine and Med...   439446   

      distance         school.type  school.main_campus  credential.level  \
673          0  Private, nonprofit                   1                 6   
13           1  Private, nonprofit                   1                 7   
654          3  Private, nonprofit                   1                 5   
297          1              Public                   1                 7   
608          1  Private, nonprofit                   1                 7   
296          1              Public                   1                 5   
677          0  Private, nonprofit                   1                 6   
858          0  Private, nonprofit                   1                 6   
1086         1  Private, nonprofit                   1                 7   
1070         1  Private, nonprofit                   1                 5   

      earnings.4_yr.overall_median_earnings  \
673                                114794.0   
13                                 106396.0   
654                                 48527.0   
297                                 82512.0   
608                                 83257.0   
296                                 49226.0   
677                                143354.0   
858                                 71477.0   
1086                                69213.0   
1070                                29614.0   

      earnings.4_yr.working_not_enrolled.overall_count  \
673                                              175.0   
13                                                88.0   
654                                               62.0   
297                                               67.0   
608                                              271.0   
296                                               39.0   
677                                              377.0   
858                                              296.0   
1086                                             116.0   
1070                                              25.0   

      debt.staff_grad_plus.all.eval_inst.median  \
673                                    406377.0   
13                                     278906.0   
654                                    125828.0   
297                                    207971.0   
608                                    202806.0   
296                                    117918.0   
677                                    336670.0   
858                                    166022.0   
1086                                   144259.0   
1070                                    61500.0   

      debt.staff_grad_plus.all.eval_inst.median_payment  \
673                                              4623.0   
13                                               3173.0   
654                                              1431.0   
297                                              2366.0   
608                                              2307.0   
296                                              1341.0   
677                                              3830.0   
858                                              1889.0   
1086                       

In [114]:
dfx.groupby("credential.level").agg({
    "earnings.4_yr.overall_median_earnings": "median",
    "debt.staff_grad_plus.all.eval_inst.median": "median",
    "debt_burden": "median",
    "unit_id": "count"
})

earnings.4_yr.overall_median_earnings  \
credential.level                                          
1                                               24719.0   
2                                               41859.5   
3                                               49000.0   
5                                               62299.0   
6                                               78641.5   
7                                               88273.0   
8                                               70409.0   

                  debt.staff_grad_plus.all.eval_inst.median  debt_burden  \
credential.level                                                           
1                                                    9500.0     0.047128   
2                                                   18784.5     0.052843   
3                                                   19250.0     0.050230   
5                                                   41000.0     0.086308   
6                                                  127406.0     0.222583   
7                                                  162455.0     0.260056   
8                                                   46692.0     0.084563   

                  unit_id  
credential.level           
1                     217  
2                     218  
3                     641  
5                     226  
6                      22  
7                      17  
8                       7

Median earnings increase with credential level; however, median debt increases disproportionately at professional levels (5 and 6), producing elevated debt burden at higher tiers.